resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/LocalFileNativesManager.java

```diff
@@ -46,16 +52,31 @@ public class LocalFileNativesManager {
 	 * reset the usage of native to the system default
 	 */
 	public static void reset() {
-		setUsingNative(Boolean.parseBoolean(System.getProperty(PROPERTY_USE_NATIVES, String.valueOf(PROPERTY_USE_NATIVE_DEFAULT))));
+		boolean useNatives = Boolean.parseBoolean(System.getProperty(PROPERTY_USE_NATIVES, String.valueOf(PROPERTY_USE_NATIVE_DEFAULT)));
+		boolean favourWindowsNative = Boolean.getBoolean(PROPERTY_NATIVES_FAVOUR_WINDOWS); // default to FALSE
+
+		setUsingNative(useNatives, favourWindowsNative);
 	}
 
 	/**
-	 * Try to set the usage of natives to the provided value
-	 * @return <code>true</code> if natives are used as result of this call <code>false</code> otherwhise
+	 * <p>Try to set the usage of natives to the provided value.</p>
+	 * 
+	 * @param useNatives TRUE if an attempt should be made to use native libraries 
+	 * by the various API methods exposed by this class 
+	 * @param favourWindowsNative TRUE if a priority preference should be given to the 
+	 * loading of a native Windows library <i>ahead of</i> a Unix native library
+	 * @return <code>true</code> if natives are used as result of this call <code>false</code> otherwise
 	 */
-	public static boolean setUsingNative(boolean useNatives) {
+	public static boolean setUsingNative(boolean useNatives, boolean favourWindowsNative) {
 		boolean nativesAreUsed;
-		if (useNatives && UnixFileNatives.isUsingNatives()) {
+
+		if (useNatives && favourWindowsNative && LocalFileNatives.isUsingNatives()) {
+			// avoids an invocation of System.loadLibrary("unixfile_1_0_0") - thereby 
```

Comment:
if useNatives && favourWindowsNative but not LocalFileNatives.isUsingNatives()
will it load System.loadLibrary("unixfile_1_0_0")?

Reply:
or (unintended) localfile_1_0_0?

Reply:
It will fall-back on the existing implementation in that case, and attempt to load _unixfile_1_0_0.dll_ via the ELSE IF block starting on line 79.

Reply:
doesnt LocalFileNatives.isUsingNatives() alread load localfile_1_0_0?

Reply:
Yes it does. 

If you are getting at the Unix OS workflow - in the case of a Unix based tool configuration (assuming a client does _not_ set the new property _-Declipse.filesystem.favourWindowsNative=true_) there will be no attempt to load the _localfile_1_0_0.dll_ (on Unix based OS) via the new condition on line 73.

For Windows OS workflow if the _-Declipse.filesystem.favourWindowsNative=true_ property is NOT set, the existing order will occur with an attempted load of _unixfile_1_0_0.dll_, followed by a load for _localfile_1_0_0.dll_.  If the _-Declipse.filesystem.favourWindowsNative=true_ property IS set, then there will be _no_ attempted load of _unixfile_1_0_0.dll_, and the first library load will be _localfile_1_0_0.dll_.